lib/navigation/imu_displacement.cpp

```diff
@@ -0,0 +1,50 @@
+#include "imu_displacement.hpp"
+
+#include <cmath>
+
+#include <chrono>
+
+namespace hyped::navigation {
+
+ImuDisplacement::ImuDisplacement()
+    : imu_displacement_(0),
+      imu_velocity_(0),
+      previous_timestamp_micros_(std::chrono::duration_cast<std::chrono::microseconds>(
+                                   std::chrono::system_clock::now().time_since_epoch())
+                                   .count())
+{
+}
+
+void ImuDisplacement::updateImuDisplacement(const core::Float imu_acceleration)
+{
+  const core::TimePoint time_now = std::chrono::system_clock::now();
+  const std::int64_t time_now_micros
+    = std::chrono::duration_cast<std::chrono::microseconds>(time_now.time_since_epoch()).count();
+  const std::int64_t time_elapsed_seconds
+    = static_cast<double>((time_now_micros - previous_timestamp_micros_) / 1'000'000);
```

Comment:
`core::Timer`

Reply:
It's the time between measurements, not the time a task takes. I'm not sure how `core::Timer` will do what I'm looking for here (unless I should just implement this functionality in `core::Timer?`)

Reply:
the latter i think

make it use the `core::Duration` and `core::TimePoint` types and take a `core::ITimeSource`